### PR TITLE
Make sure that only valid middlewares are accepted

### PR DESCRIPTION
**Pull request by rtuin**
This should cover the issue found in https://github.com/thephpleague/tactician/issues/54

### CODE REVIEW

File: tests/CommandBusTest.php [hunk 1]
```diff
@@ -67,4 +68,24 @@ public function testHandleThrowExceptionForInvalidCommand()
...
     }
+
+    public function testIfOneCanOnlyCreateWithValidMiddlewares()
+    {
+        $middlewareList = [$this->getMock('stdClass')];
+
+        try {
+            new CommandBus($middlewareList);
+            $this->fail('Should not be able to create a CommandBus without a valid list of middlewares!');
+        } catch (InvalidMiddlewareException $caught) {
```
Comment (boekkooi) [inline tests/CommandBusTest.php]: There is no need to use a try catch seehttps://phpunit.de/manual/current/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions also using a dataprovider to test for a string or null maybe better

Comment (rtuin) [inline tests/CommandBusTest.php]: This is how I like to test for expected exceptions, as this allows me to present a meaningful message when the exception does not occur.

Comment (boekkooi) [inline tests/CommandBusTest.php]: It is still possible since the exception check is done once the is finished.
```PHP
$this->setExpectedException('League\Tactician\Exception\InvalidMiddlewareException');

$middlewareList = [$this->getMock('stdClass')];
new CommandBus($middlewareList);
$this->fail('Should not be able to create a CommandBus without a valid list of middlewares!');
```

File: src/Exception/InvalidMiddlewareException.php [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace League\Tactician\Exception;
+
+class InvalidMiddlewareException extends \LogicException implements Exception
```
Comment (rosstuck) [inline src/Exception/InvalidMiddlewareException.php]: Needs a little docblock and maybe `InvalidArgumentException` is a better base class here?

File: src/Exception/InvalidMiddlewareException.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+{
+    public static function forMiddleware($middleware)
+    {
+        $name = is_object($middleware) ? get_class($middleware) : gettype($middleware);
```
Comment (rosstuck) [inline src/Exception/InvalidMiddlewareException.php]: Good move. :)

File: tests/CommandBusTest.php (hunk 1 above)
Comment (rosstuck) [inline tests/CommandBusTest.php]: Huh, TIL.

The exception itself is already fairly meaningful so just the expectedException annotation is enough for me here. If you'd like to be more specific, either format is fine with me.

**Comment by rosstuck** [comment]
Gimmie a sign when the comments are addressed and I'll gladly merge this one. Thanks a ton! :)

File: tests/CommandBusTest.php (hunk 1 above)
Comment (rtuin) [inline tests/CommandBusTest.php]: Hah, interesting indeed! I cleaned up the test method a bit accordingly :+1:

**Comment by rtuin** [comment]
@rosstuck 
image

**Comment by rosstuck** [comment]
Thank you!
